frontend/src/components/PointPopup/PointPopup.tsx

```diff
@@ -22,36 +14,130 @@ import {
-  return <IconButton {...other} />;
-})(({ theme, expand }) => ({
-  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
-  marginLeft: 'auto',
-  transition: theme.transitions.create('transform', {
-    duration: theme.transitions.duration.shortest
-  })
-}));
+  const { updatePopupLocation } = useTypedDispatch();
+
+  const userLogin = useTypedSelector(state => state.userLogin);
+  const infoLocation = useTypedSelector(state => state.popupLocation);
+
+  const {
+    isLogged,
+    userInfo: { user }
+  } = userLogin;
+
+  const { _id: userId } = user;
+  const { _id, rating, locationName, description, arrayPhotos } = infoLocation;
+
+  const handleCloseNotification = (
+    e?: SyntheticEvent | Event,
+    reason?: string
+  ) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setNotification(null);
+  };
 
-function PointPopup({ location }: Props) {
-  const [expanded, setExpanded] = useState(false);
   const handleExpandClick = () => {
     setExpanded(!expanded);
   };
 
+  const handleLikes = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (!isLogged) {
+      return setNotification({
+        type: 'warning',
+        message: 'Login to your account to be able to likes!'
+      });
+    }
+    const updatedRating = { ...rating };
+    if (rating.likes.includes(userId)) {
+      updatedRating.likes = updatedRating.likes.filter(
+        value => value !== userId
+      );
+    } else {
+      updatedRating.likes.push(userId);
+    }
+
+    if (rating.dislikes.includes(userId)) {
+      updatedRating.dislikes = updatedRating.dislikes.filter(
+        value => value !== userId
+      );
+    }
+
+    return updatePopupLocation(_id, { rating: updatedRating });
+  };
+
+  const handleDislikes = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (!isLogged) {
+      return setNotification({
+        type: 'warning',
+        message: 'Login to your account to be able to likes!'
+      });
+    }
+    const updatedRating = { ...rating };
+    if (rating.dislikes.includes(userId)) {
+      updatedRating.dislikes = updatedRating.dislikes.filter(
+        value => value !== userId
+      );
+    } else {
+      updatedRating.dislikes.push(userId);
+    }
+
+    if (rating.likes.includes(userId)) {
+      updatedRating.likes = updatedRating.likes.filter(
+        value => value !== userId
+      );
+    }
+
+    return updatePopupLocation(_id, { rating: updatedRating });
+  };
+
+  let snackbar;
+  if (notification) {
+    snackbar = (
+      <Snackbar
+        anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
+        open={!!notification}
+        autoHideDuration={5000}
+        onClose={handleCloseNotification}
+      >
+        <Alert
+          onClose={handleCloseNotification}
+          severity={notification.type}
+          sx={{ width: '100%' }}
+        >
+          {notification.message}
+        </Alert>
+      </Snackbar>
+    );
+  }
```

Comment:
should be moved in useEffect on notification change!

Reply:
We don't need it at all. Useeffect is not used for this